libr/anal/fcn.c

```diff
@@ -344,6 +344,31 @@ static RAnalBlock *bbget(RAnalFunction *fcn, ut64 addr) {
 	return NULL;
 }
 
+// Use this function if anal.jmpmid == true
+static RAnalBlock **bbget_all(RAnalFunction *fcn, ut64 addr) {
+	RListIter *iter;
+	RAnalBlock *bb;
+	// TODO: remove need for malloc
+	RAnalBlock **bbs = malloc (sizeof (RAnalBlock *) * (r_list_length (fcn->bbs) + 1));
+	RAnalBlock **ptr;
+	r_return_val_if_fail (bbs, NULL);
```

Comment:
no please. See https://github.com/radare/radare2/blob/master/libr/include/r_util/r_assert.h#L68 . `r_return_val_if_fail` should not be used for runtime errors and a failing malloc is a runtime error, something that could happen. Just use the usual `if (!bbs) return NULL`

Reply:
ok a571473e63e26d036a86bed0a8c6c87d86d21dd0